yt_dlp/extractor/goplay.py

```diff
@@ -0,0 +1,404 @@
+        },
+        'skip': 'This video is only available for registered users'
+    }]
+
+    def _initialize_pre_login(self):
+        self._id_token = None
+
+    def _perform_login(self, username, password):
+        self.report_login()
+        aws = AwsIdp(pool_id='eu-west-1_dViSsKM5Y', client_id='6s1h851s8uplco5h6mqh1jac8m')
+        self._id_token, _ = aws.authenticate(username=username, password=password)
+
+    def _real_initialize(self):
+        if not self._id_token:
+            raise self.raise_login_required(method='password')
+
+    def _real_extract(self, url):
+        display_id = url.split('#')[0].split('/')[-1]
+        webpage = self._download_webpage(url, display_id)
+        video_data_json = self._html_search_regex(r'<div\s+data-hero="([^"]+)"', webpage, 'video_data')
+        video_data = self._parse_json(video_data_json.replace('quot;', '"'), display_id)['data']
+
+        movie = video_data['movie']
+        if movie:
+            video_id = movie['videoUuid']
+            data = {
+                'title': movie.get('title')
+            }
+        else:
+            playlists = video_data['playlists']
+            episode = traverse_obj(
+                traverse_obj(playlists, (lambda _, v: v['pageInfo']['url'] in url, 'episodes'), get_all=False),
+                lambda _, v: v['pageInfo']['url'] == url, lambda _, v: v['pageInfo']['url'] in url, get_all=False)
+            video_id = episode['videoUuid']
+            data = {
+                'title': episode.get('episodeTitle'),
+                'series': traverse_obj(episode, ('program', 'title')),
+                'season_number': episode.get('seasonNumber'),
+                'episode_number': episode.get('episodeNumber'),
+            }
+
+        api = self._download_json(
+            f'https://api.viervijfzes.be/content/{video_id}',
+            video_id, headers={'Authorization': self._id_token})
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            api['video']['S'], video_id, ext='mp4', m3u8_id='HLS')
+        self._sort_formats(formats)
+
+        data.update({
+            'id': video_id,
+            'formats': formats,
+        })
+
+        return data
+
+
+# Taken from https://github.com/add-ons/plugin.video.viervijfzes/blob/master/resources/lib/viervijfzes/auth_awsidp.py
+# with some small modifications (replacement of six and requests, removal of unused function, reformatting)
+# This file is licensed as GPL-3 from https://github.com/add-ons/plugin.video.viervijfzes:
+# https://github.com/add-ons/plugin.video.viervijfzes/blob/master/LICENSE
+
+# Amazon Cognito implementation without external dependencies
+# Based on https://github.com/retrospect-addon/plugin.video.retrospect/blob/master/channels/channel.be/vier/awsidp.py
+
+import base64
+import binascii
+import datetime
+import hashlib
+import hmac
+import json
+import os
+
+from ..compat import compat_basestring, compat_urllib_request
```

Comment:
Replacing `compat_basestring = str` will fix the deprecation warning. But I recommend to replace `compat_urllib_request = urllib.request` as well

PS: Move all imports to top of file